build/configure.js

```diff
@@ -0,0 +1,294 @@
+'use strict';
+// Generate and execute makefile
+
+var utils = require('utils');
+
+var PRE_APP_BUILD_STEPS = [
+  'clean-stage-app',
+  'svoperapps',
+  'webapp-manifests',
+  'contacts-import-services',
+  'search-provider',
+  'keyboard-layouts',
+  'preferences',
+  'settings',
+  'webapp-shared',
+  'copy-common-files'
+];
+
+var POST_APP_BUILD_STEPS = [
+  'media-resolution',
+  'post-manifest',
+  'multilocale',
+  'copy-build-stage-data',
+  'webapp-optimize',
+  'webapp-zip'
+];
+
+var Makefile = function(filename) {
```

Comment:
Could we put these 'classes' in individual files, instead of put them all in the same file? We may require them after you did that.